### PR TITLE
pinentry: 1.0.0 → 1.1.0

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
https://github.com/gpg/pinentry/blob/pinentry-1.1.0/NEWS#L1-L19

###### Things done

I ran pinentry binary from each of the package end tested `GETPIN` command.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jtojnar** [comment]
@GrahamcOfBorg build pinentry pinentry_ncurses pinentry_gnome pinentry_qt4 pinentry_qt5

File: nixos/modules/config/no-x-libs.nix [hunk 1]
```diff
@@ -35,7 +35,7 @@ with lib;
       networkmanager_pptp = pkgs.networkmanager_pptp.override { withGnome = false; };
       networkmanager_vpnc = pkgs.networkmanager_vpnc.override { withGnome = false; };
       networkmanager_iodine = pkgs.networkmanager_iodine.override { withGnome = false; };
-      pinentry = pkgs.pinentry.override { gtk2 = null; qt4 = null; };
+      pinentry = pkgs.pinentry.override { gtk2 = null; };
```
Comment (orivej) [inline nixos/modules/config/no-x-libs.nix]: Just set `pinentry = pkgs.pinentry_ncurses`.

Comment (jtojnar) [reply nixos/modules/config/no-x-libs.nix]: Won’t that be infinite recursion?

Comment (orivej) [reply nixos/modules/config/no-x-libs.nix]: Right, I'd define `pinentry` as an override of `pinentry_ncurses` in `all-packages.nix` to solve it.

Comment (jtojnar) [reply nixos/modules/config/no-x-libs.nix]: Right

**Review by orivej — APPROVED** [review]

**Pull request merged**